app/V1Module/presenters/SubmitPresenter.php

```diff
@@ -268,11 +269,21 @@ private function finishSubmission(AssignmentSolution $solution, bool $isDebug =
 
     // generate job configuration
     $compilationParams = CompilationParams::create($solution->getSolution()->getFileNames(), $isDebug, $solution->getSolution()->getSolutionParams());
-    $generatorResult =
-      $this->jobConfigGenerator->generateJobConfig($this->getCurrentUser(),
-        $solution->getAssignment(),
-        $solution->getSolution()->getRuntimeEnvironment(),
-        $compilationParams);
+
+    try {
+      $generatorResult =
+        $this->jobConfigGenerator->generateJobConfig($this->getCurrentUser(),
+          $solution->getAssignment(),
+          $solution->getSolution()->getRuntimeEnvironment(),
+          $compilationParams);
+    } catch (ExerciseConfigException | JobConfigStorageException $e) {
+      $submission = new AssignmentSolutionSubmission($solution, "", $this->getCurrentUser());
+      $this->assignmentSubmissions->persist($submission, false);
+      $failure = SubmissionFailure::forSubmission(SubmissionFailure::TYPE_CONFIG_ERROR, $e->getMessage(), $submission);
+      $this->submissionFailures->persist($failure);
+      $this->failureHelper->report(FailureHelper::TYPE_API_ERROR, "Failed to generate job config for {$submission->getId()}");
+      throw new SubmissionFailedException($e->getMessage());
+    }
```

Comment:
I have to say, that I don't really like this, it is duplicit and easy to forget if something changes. Also if you do this you also have to solve this in reference submissions...

Reply:
What about using `submissionFailed` method?

Reply:
We'd have to add a parameter or two, but it would be a better solution, yes. Reference submissions work a little differently, but I also checked there.

Reply:
If it would be better, let's do that! Just a quick quesch, what is different in reference submissions?

Reply:
Reference submissions are typically added by people who know that something can (and oftend does) go wrong. The response also contains a field for errors that we can fill.

Reply:
KK... so lets change only failure helper method in normal submit...